### ISSUE TITLE
MSER Connected Components not accessable from class

### ISSUE THREAD

**Issue by guysoft** [Itseez/opencv#6575]
### Please state the information for your system
- OpenCV version: 3.x
- Host OS: Linux (Ubuntu 16.04)

### In which part of the OpenCV library you got the issue?
- MSER feature detector

### Expected behaviour

I would like to have access to the connected components from MSER algorithm in order to implement text detection.

### Actual behaviour
There is not way to get the connected components

### Additional description
Hey, I want to implement [robust text detection in natural images](http://web.stanford.edu/~hchen2/papers/ICIP2011_RobustTextDetection.pdf), as shown in this [official matlab example](http://www.mathworks.com/help/vision/examples/automatically-detect-and-recognize-text-in-natural-images.html). But in order to do it you must have access to the connected components within the MSER algorithm.
Those are [computed by opencv here](https://github.com/Itseez/opencv/blob/09df1a286b62450f2b619ee1d4fec02000daa2e4/modules/features2d/src/mser.cpp#L368), but as you can see from the code they are not saved anywhere and there is no way to pull them out of the function. I am not sure how to pull this from the c++ library without breaking something. I would gladly make a pull request if someone points me in the right direction.

### Code example to reproduce the issue / Steps to reproduce the issue
Please try to give a full example which will compile as is.

python:
```python
    img = cv2.imread(img)
    mser.setDelta(4)
    mser.setMinArea(200)
    mser.setMaxArea(8000)
    ngrey = cv2.cvtColor(origin, cv2.COLOR_BGR2GRAY)
    regions = mser.detectRegions(ngrey, None)
    #would like to have
    #regions, mser_comp = mser.detectRegionsWithConnectedComponents(img, None)

```
c++:
```c++
Mat image = imread( img);
    vector<vector<Point>> contours;
    Ptr<MSER> mserExtractor  = MSER::create();
    vector<vector<Point>> contours;
   cvtColor( image, grey, CV_BGR2GRAY );
   vector<cv::Rect> mserBbox;
    mserExtractor->detectRegions(grey, contours, mserBbox);
```

**Comment by schenker** [comment]
@guysoft i have the same problem. Did you find a solution yet?

**Comment by guysoft** [comment]
@schenker no, I haven't I found someone trying to implement the [connected components in c++ but its not efficient enough here](https://github.com/rbgirshick/py-faster-rcnn).

**Comment by bpawlik** [comment]
@guysoft  API change is a big step, do you necessarily need to have it as an OpenCV component? If you're still interested I can create a modification for you in my fork from which you can pull, but won't recommend to merge it with master.

**Comment by guysoft** [comment]
I don't mind forking, I am pretty good with git and can maintain such a
fork.

Also, my need for this is not as urgent as before, but it might speed
things up for what I am working on.
I'm just not sure how MSER is working form end to end, which makes it hard
for me to code that bit.

**Comment by bpawlik** [comment]
According to the matlab code that you refered to, you need to get access to the connected components which structure looks as follows: http://www.mathworks.com/help/vision/ref/detectmserfeatures.html#outputarg_cc
I believe it's PixelIdxList that you're mostly interested in? In that case, why don't you use vector of vectors of Points (msers argument of MSER::detectRegions)?
That ConnectedComponents struct in mser.cpp is a linked list that won't give you any more information than msers (or contours as you refer in your example). Hope this helps.
